package/kedro_viz/models/flowchart.py

```diff
@@ -804,6 +751,25 @@ def set_preview(cls, _):
             )
             return None
 
+    @validator("preview_type", always=True)
+    def set_preview_type(cls, _):
+        if not (
+            not cls.data_node.disable_preview() and hasattr(cls.dataset, "preview")
```

Comment:
shouldn't this check for `preview_type` instead of `preview`?
```suggestion
            not cls.data_node.disable_preview() and hasattr(cls.dataset, "preview_type")
```

Reply:
The dataset/datanode only has a preview function. `preview_type` is derived from the signature.